optimade/validator/validator.py

```diff
@@ -147,14 +151,16 @@ def __init__(  # pylint: disable=too-many-arguments
         self._test_id_by_type = {}
         self._entry_info_by_type = {}
 
-        self.success_count = 0
-        self.failure_count = 0
-        self.internal_failure_count = 0
-        self.optional_success_count = 0
-        self.optional_failure_count = 0
-        self.failure_messages = []
-        self.internal_failure_messages = []
-        self.optional_failure_messages = []
+        self.results = {
```

Comment:
This is nice.
It could even be a separate class that holds all these keys as attributes, but no need to overcomplicate it of course :)

Reply:
Agreed, I started doing exactly this with a named tuple but it felt a little overengineered given that its only used in once place...

Reply:
I might actually switch back to the namedtuple if I can find a nice way to get a JSON dump of it without having to write any extra methods